parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/HadoopPositionOutputStream.java

```diff
@@ -62,7 +64,9 @@ public void flush() throws IOException {
   @Override
   public void close() throws IOException {
     try (FSDataOutputStream fdos = wrapped) {
```

Comment:
looking at this more, while it does guarantee wrapped.close() is called and is very pretty, there's a risk that if hflush() raises an exception then anything raised by wrapped.close() is lost. 
when working to s3, it's that close() where the write to S3 takes place and is where the more important failures are likely to be raised.
It might be best to catch and log the hflush failure and always call wrapped.close()